### ISSUE TITLE
Aligned runs in time-based pagination may drop some runs

### ISSUE THREAD

**Issue by lukebjerring** [web-platform-tests/wpt.fyi#819]
https://github.com/web-platform-tests/wpt.fyi/pull/816 introduced pagination via to/from params.

In the case that we are looking for aligned runs, where not all runs occur within the same page, the runs will not be present in either of the pages.

**Comment by lukebjerring** [comment]
@foolip - the easiest, but inefficient, way to handle this would be to ignore the `to` param during the process of querying Datastore for the keys, and instead use it to filter the loaded runs (when the earliest of the runs for a given SHA is not inside the time range).

I'm not super happy about that though; it becomes increasingly more expensive the further you paginate into the data. We could mitigate the expense by running one extra query for the previous page's time range only, which should be accurate for large enough time ranges.

It seems inherently problematic to do things this way. Perhaps the time stamp should be that of the commit in wpt instead of the start-time of the run?

**Comment by foolip** [comment]
Yep, using commit time would fix, as well as the occasional incorrect order that we currently have.
